src/Compilers/CSharp/Portable/FlowAnalysis/NullableWalker.cs

```diff
@@ -797,7 +831,9 @@ private void InheritNullableStateOfMember(int targetContainerSlot, int valueCont
             if (fieldOrPropertyType.IsReferenceType || fieldOrPropertyType.IsNullableType())
             {
                 int targetMemberSlot = GetOrCreateSlot(member, targetContainerSlot);
-                NullableAnnotation value = fieldOrPropertyType.NullableAnnotation;
+                NullableAnnotation value = (isDefaultValue && fieldOrPropertyType.IsReferenceType) ?
```

Comment:
>fieldOrPropertyType.IsReferenceType [](start = 62, length = 35)

Should this be `... && (fieldOrPropertType.IsValueType || fieldOrProperty.IsNullableType())`?

Reply:
If the member type is a nullable value type, `fieldOrPropertyType.NullableAnnotation` should return `Nullable` already.

---
In reply to: [246866495](https://github.com/dotnet/roslyn/pull/32304#discussion_r246866495) [](ancestors = 246866495)

Reply:
A member with a unconstrained type parameter type is already considered maybe null. The only cases we need to handle specifically here are members where the type is declared as not nullable.

---
In reply to: [246871048](https://github.com/dotnet/roslyn/pull/32304#discussion_r246871048) [](ancestors = 246871048,246867516,246866495)

Reply:
There is a related issue when copying not-nullable state for unconstrained types though. Logged [https://github.com/dotnet/roslyn/issues/32338](https://github.com/dotnet/roslyn/issues/32338). Thanks.

---
In reply to: [246874741](https://github.com/dotnet/roslyn/pull/32304#discussion_r246874741) [](ancestors = 246874741,246871048,246867516,246866495)